### PR TITLE
[KOGITO-4956] avoid registering anonymous rule units

### PR DESCRIPTION
**Pull request by mariofusco**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by mariofusco** [comment]
@evacchi @danielezonca I didn't remove the `unitRegistry` map, but only populating it when a name is explicitly given instead of generating an `UUID` that anyway is not reachable as I did before.

This allowed me to also keep [this test](https://github.com/kiegroup/kogito-runtimes/blob/4aa991409dab43747e55819370cd9a934d98cbd4/kogito-codegen-modules/kogito-codegen-integration-tests/src/test/java/org/kie/kogito/codegen/RuleUnitCompilerTest.java#L172) that is the only one using that registry so far and also demonstrating how I intended to use it. If you think that this usage pattern is not useful or anyway in contrast with the direction of the other APIs we're implementing please let me know and I'll remove both the registry and the test. At the moment I'm not strongly opinionated about this.

**Review by evacchi — APPROVED** [review]

**Comment by evacchi** [comment]
well @mariofusco that won't eliminate the "leak" unless we provide some sort of "remove/dispose" method. Something we need to be aware of in the future.

File: drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/impl/AbstractRuleUnit.java [hunk 1]
```diff
@@ -41,7 +41,9 @@ public String id() {
     @Override
     public RuleUnitInstance<T> createInstance(T data, String name) {
         RuleUnitInstance<T> instance = internalCreateInstance(data);
-        app.get(RuleUnits.class).register(name, instance);
+        if (name != null) {
```
Comment (danielezonca) [inline drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/impl/AbstractRuleUnit.java:44]: What about add this null check inside `AbstractRuleUnits.register` (maybe logging error or throwing an exception as you prefer)?

Comment (mariofusco) [reply drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/impl/AbstractRuleUnit.java:44]: @danielezonca do you mean adding the null check **also** there (for defensive reasons) and throwing an exception if it is called with a null? I cannot remove the null check here, correct?

Comment (danielezonca) [reply drools/kogito-ruleunits/src/main/java/org/kie/kogito/rules/units/impl/AbstractRuleUnit.java:44]: Yes I mean in addition to this code.
`register` method is public so it is better to be on the safe side

**Review by danielezonca — APPROVED** [review]

**Pull request merged**